query/stdlib/influxdata/influxdb/from_test.go

```diff
@@ -145,3 +149,703 @@ func TestFromOpSpec_BucketsAccessed(t *testing.T) {
+					plan.CreatePhysicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeExcept,
+						GroupKeys: []string{"_time", "_value"},
+					}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+				},
+			},
+			NoChange: true,
+		},
+		{
+			Name: "from group _time",
+			// We should not push down group(columns: ["_time"])
+			Rules: []plan.Rule{influxdb.MergeFromGroupRule{}},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.PlanNode{
+					plan.CreatePhysicalNode("from", from),
+					plan.CreatePhysicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeExcept,
+						GroupKeys: []string{"_time"},
+					}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+				},
+			},
+			NoChange: true,
+		},
+		{
+			Name: "from group _value",
+			// We should not push down group(columns: ["_value"])
+			Rules: []plan.Rule{influxdb.MergeFromGroupRule{}},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.PlanNode{
+					plan.CreatePhysicalNode("from", from),
+					plan.CreatePhysicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeExcept,
+						GroupKeys: []string{"_value"},
+					}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+				},
+			},
+			NoChange: true,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			plantest.RuleTestHelper(t, &tc)
+		})
+	}
+}
+
+func TestFromKeysRule(t *testing.T) {
+	tests := []plantest.RuleTestCase{
+		{
+			Name:  "from keys",
+			Rules: []plan.Rule{influxdb.FromKeysRule{}},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.PlanNode{
+					plan.CreatePhysicalNode("from", &influxdb.FromProcedureSpec{}),
+					plan.CreatePhysicalNode("keys", &universe.KeysProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.PlanNode{
+					plan.CreatePhysicalNode("from", &influxdb.FromProcedureSpec{
+						LimitSet:    true,
+						PointsLimit: -1,
+					}),
+					plan.CreatePhysicalNode("keys", &universe.KeysProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			plantest.RuleTestHelper(t, &tc)
+		})
+	}
+}
+
+func TestFromRangeValidation(t *testing.T) {
+	testSpec := plantest.PlanSpec{
+		//       3
+		//     /  \
+		//    /    \
+		//   1      2
+		//    \    /
+		//     from
+		Nodes: []plan.PlanNode{
+			plan.CreatePhysicalNode("from", &influxdb.FromProcedureSpec{}),
+			plantest.CreatePhysicalMockNode("1"),
+			plantest.CreatePhysicalMockNode("2"),
+			plantest.CreatePhysicalMockNode("3"),
+		},
+		Edges: [][2]int{
+			{0, 1},
+			{0, 2},
+			{1, 3},
+			{2, 3},
+		},
+	}
+
+	ps := plantest.CreatePlanSpec(&testSpec)
+	pp := plan.NewPhysicalPlanner(plan.OnlyPhysicalRules())
+	_, err := pp.Plan(ps)
+
+	if err == nil {
+		t.Error("Expected from with no range to fail physical planning")
+	}
+}
```

Comment:
I'm also assuming that this code was copied in a straightforward manner from the `flux` repo.

Reply:
Same as above :)